monai/transforms/utils.py

```
@@ -2098,9 +2099,18 @@ def scale_affine(spatial_size, new_spatial_size, centered: bool = True):
     r = max(len(new_spatial_size), len(spatial_size))
     if spatial_size == new_spatial_size:
         return np.eye(r + 1)
-    s = np.array([float(o) / float(max(n, 1)) for o, n in zip(spatial_size, new_spatial_size)], dtype=float)
+    if align_corners:
+        # Match interpolate behavior: (src-1)/(dst-1)
+        s = np.array(
+            [(float(o) - 1) / max(float(n) - 1, 1) for o, n in zip(spatial_size, new_spatial_size)], dtype=float
```

Ruff (0.15.2)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)